src/parser/priest/discipline/normalizers/PowerWordRadianceNormalizer.ts

```diff
@@ -16,8 +16,8 @@ const MAX_TIME_SINCE_CAST = 250; // ms
  events so that the applications are always right after the cast.
  */
 class PowerWordRadianceNormalizer extends EventsNormalizer {
-  normalize(events) {
-    const fixedEvents = [];
+  normalize(events: Array<ApplyBuffEvent | RefreshBuffEvent | ApplyBuffStackEvent | CastEvent>) {
```

Comment:
This looks at all events, its `Event<any>` (same with fixedEvents declaration)

Reply:
The base event doesn't have ability: Ability which is what is being used in this module. That's why I manually added the possible types.

Reply:
I have already added a type guard for this exact sitch! 
```typescript
     if(!HasAbility(event)) {
        return;
      }
```
Can add that right after the first `fixedEvents.push(event)` and all of the other instances of event.ability shouldn't error. Your casting is also fine in this case.

The reason this is important is because by telling the method that the arguments are of a certain type, the scope of the function will take your word as law. The way you had it originally, you could technically write `console.log(event.ability.guid)` on L20, before any of the `event.type` checks, and typescript would say that's fine. However, because `normalize` actually looks at all events, not only ability events, you would always get a crash that says `event.ability` is undefined when you try to access `guid`.

Technically resolved conversation, I'll leave casting vs typeguard up to you